miopengemm/src/alphagenerator.cpp

```diff
@@ -788,7 +788,15 @@ const TSHORT group_id_z = group_id % N_WORK_ITEMS_PER_C_ELM;
       ss << "\n/* branching between work groups : some wgs have 1 more unroll "
             "to process. */\n";
       ss << "int n_unrolls_remaining = (" << dp.k_effective_div_G_UNROLL;
-      ss << ") +  (group_id_z < n_work_groups_with_1_more);";
+      ss << ") ";
+
+     // To avoid the compiler outsmarting us when n_work_groups_with_1_more is zero 
+     // [tautological-unsigned-zero-compare], we avoid checking if unsigned integers
+     // are less than zero. 
+     if (dp.k_effective_mod_G_UNROLL > 0){
```

Comment:
">" or ">="?

Reply:
`dp.k_effective_mod_G_UNROLL` is the initializer of `n_work_groups_with_1_more`. If it equals to 0, then the value of `(group_id_z < n_work_groups_with_1_more)` == `(group_id_z < 0)` == 0, which is useless. And unsigned comparison leads to warning.

Reply:
Does the situation that dp.k_effective_mod_G_UNROLL=0 happen then? @atamazov

Reply:
Yes, that is what happens in https://github.com/ROCmSoftwarePlatform/MIOpen/issues/692